lib/trino-hdfs/src/main/java/io/trino/hdfs/TrinoFileSystemCache.java

```diff
@@ -399,6 +401,23 @@ public RemoteIterator<LocatedFileStatus> listFiles(Path path, boolean recursive)
         {
             return fs.listFiles(path, recursive);
         }
+
+        @Override
+        public void batchDeleteFiles(Collection<Path> paths)
+                throws IOException
+        {
+            FileSystem rawFileSystem = getRawFileSystem();
```

Comment:
I hate the fact that I am doing similar thing twice but I didn't see a way to avoid this repetition..